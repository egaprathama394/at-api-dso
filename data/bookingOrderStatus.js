export default {
  orderDaimoCode: 'DFA202000000123',
  pssOrderStatus: '2',
};
