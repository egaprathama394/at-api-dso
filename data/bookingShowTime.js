export default {
  branchCode: 'D004',
  scheduleDate: '2020-09-30 00:00:00',
  vehicleType: 'XENIA MT',
  jobs: [
    {
      jobCode: 'OTHER',
      jobDescription: 'PERBAIKAN LAINNYA',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};

export const negativeRequest = {
  branchCode: 'D004',
  scheduleDate: '2018-10-26 00:00:00',
  vehicleType: 'XENIA MT',
  jobs: [
    {
      jobCode: 'OTHER',
      jobDescription: 'PERBAIKAN LAINNYA',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};
