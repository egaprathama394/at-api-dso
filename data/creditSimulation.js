export default {
  P_NEW_USED: 'N',
  P_CATEGORY: 'EX2',
  P_VEHICLE_CATEGORY: '1',
  P_BRAND: '002',
  P_TYPE: '1',
  P_MODEL: '1',
  P_DP: '49125000',
  P_OTR: '196500000',
  P_TYPE_INSU: 'A',
  P_MODE_INSU: 'K',
  P_FLAG_ACP: 'Y',
  P_TENORVALUE: '12',
};