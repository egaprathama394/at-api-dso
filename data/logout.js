export default {
  applicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
  deviceID: '8B55A4DA-5331-47F9-9C8D-838A110FA18C',
};


export const negativeRequest = {
  action: 'Display',
  applicationName: '9093D285-5263-4B9A-97D7-C185189B0695',
  userVersion: '1.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};
