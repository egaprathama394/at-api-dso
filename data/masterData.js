
export default {
  companyCode: '0003',
  masterDataInfo: [
    {
      lastSync: '1900-01-01 00:00:00',
      objectName: 'Branch',
    },
  ],
  vehicleBrandCode: 'DAIHATSU',
  accountType: '1',
  action: '',
  deviceType: '0',
  operatingSystem: '0',
  serVersion: '1.0.0',
};

export const JobServiceCompany = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'JobService_Company',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',

};

export const VehicleBrand = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'vehicleBrand',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};

export const district = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'district',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',

};

export const province = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'province',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',

};

export const region = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'region',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',

};

export const holidayCalendar = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'holidayCalendar',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',

};

// looping data for job service company

export const loopVehicleType = [
  {
    vehicleType: 'AYLA',
    title: 'test case get master data job service company where vehicle = AYLA',
  },
  {
    vehicleType: 'GRANMAX',
    title: 'test case get master data job service company where vehicle = GRANMAX',
  },
  {
    vehicleType: 'GRANMAX1.3',
    title: 'test case get master data job service company where vehicle = GRANMAX1.3',
  },
  {
    vehicleType: 'GRANMAX1.5',
    title: 'test case get master data job service company where vehicle = GRANMAX1.5',
  },
  {
    vehicleType: 'LUXIO',
    title: 'test case get master data job service company where vehicle = LUXIO',
  },
  {
    vehicleType: 'Sigra 1.0',
    title: 'test case get master data job service company where vehicle = Sigra 1.0',
  },
  {
    vehicleType: 'Sigra 1.2',
    title: 'test case get master data job service company where vehicle = Sigra 1.2',
  },
  {
    vehicleType: 'SIRION',
    title: 'test case get master data job service company where vehicle = SIRION',
  },
  {
    vehicleType: 'TERIOS',
    title: 'test case get master data job service company where vehicle = TERIOS',
  },
  {
    vehicleType: 'XENIA1.0',
    title: 'test case get master data job service company where vehicle = XENIA1.0',
  },
  {
    vehicleType: 'XENIA1.3',
    title: 'test case get master data job service company where vehicle = XENIA1.3',
  },
  {
    vehicleType: 'OTHERS',
    title: 'test case get master data job service company where vehicle = OTHERS',
  },
];

export const VehicleTestDrive = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'VehicleTestDrive',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};

export const parameterConfigApp = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'ParameterConfigApplication',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};
