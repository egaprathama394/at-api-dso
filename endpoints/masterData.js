/* eslint no-unused-vars: off */
/* global env */
const supertest = require('supertest');
const env = require('dotenv').config();

process.env.NODE_TLS_REJECT_UNAUTHORIZED = 0;

const api = supertest(process.env.API_BASE_URL_MOBILE);

export default body => api.post('/api/masterdatainfo')
  .set('ClientTag', process.env.CLIENT_TAG)
  .set('server-code', process.env.SERVER_CODE_MOBILE)
  .send(body);
