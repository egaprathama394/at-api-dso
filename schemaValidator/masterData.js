export default {
  properties: {
    acknowledge: {
      type: 'number',
    },
    branch: {
      type: 'array',
      properties: {
        id: {
          type: 'number',
        },
        companyCode: {
          type: 'string',
        },
        regionName: {
          type: 'string',
        },
        branchCode: {
          type: 'string',
        },
        branchTypeCode: {
          type: 'string',
        },
        branchDescription: {
          type: 'string',
        },
        branchPICName: {
          type: 'string',
        },
        bookType: {
          type: 'number',
        },
        postalCode: {
          type: 'string',
        },
        address: {
          type: 'string',
        },
        latitude: {
          type: 'number',
        },
        longitude: {
          type: 'number',
        },
        distance: {
          type: 'number',
        },
        phone1: {
          type: 'string',
        },
        phone2: {
          type: 'string',
        },
        fax: {
          type: 'string',
        },
        branchEmailTo: {
          type: 'string',
        },
        branchEmailCc: {
          type: 'string',
        },
        webSiteLink: {
          type: 'string',
        },
        mondayToFriday: {
          type: 'string',
        },
        saturday: {
          type: 'string',
        },
        sunday: {
          type: 'string',
        },
        isMainDealer: {
          type: 'boolean',
        },
        isActive: {
          type: 'boolean',
        },
        districtId: {
          type: 'number',
        },
        districtName: {
          type: 'string',
        },
        hasBookingStall: {
          type: 'boolean',
        },
      },
      required: [
        'id',
        'companyCode',
        'regionName',
        'branchCode',
        'branchTypeCode',
        'branchDescription',
        'branchPICName',
        'bookType',
        'postalCode',
        'address',
        'latitude',
        'longitude',
        'distance',
        'phone1',
        'phone2',
        'fax',
        'branchEmailTo',
        'branchEmailCc',
        'webSiteLink',
        'mondayToFriday',
        'saturday',
        'sunday',
        'isMainDealer',
        'isActive',
        'districtId',
        'districtName',
        'hasBookingStall',
      ],
    },
    build: {
      type: 'string',
    },
    district: {
      type: 'array',
    },
    hobbies: {
      type: 'array',
    },
    holidayCalendar: {
      type: 'array',
    },
    jobService: {
      type: 'array',
    },
    jobServiceCompany: {
      type: 'array',
    },
    parameterConfigApplication: {
      type: 'array',
    },
    province: {
      type: 'array',
    },
    region: {
      type: 'array',
    },
    vehicleBrand: {
      type: 'array',
    },
    vehicleTestDrive: {
      type: 'array',
    },
    version: {
      type: 'string',
    },
    location: {
      type: 'array',
    },
  },
  required: [
    'acknowledge',
    'branch',
    'build',
    'district',
    'hobbies',
    'holidayCalendar',
    'jobService',
    'jobServiceCompany',
    'parameterConfigApplication',
    'province',
    'region',
    'vehicleBrand',
    'vehicleTestDrive',
    'version',
    'location',
  ],
  additionalProperties: false,
};

export const jobServiceCompanySchema = {
  type: 'object',
  required: [
    'branch',
    'region',
    'jobService',
    'jobServiceCompany',
    'vehicleBrand',
    'holidayCalendar',
    'parameterConfigApplication',
    'hobbies',
    'province',
    'district',
    'vehicleTestDrive',
    'location',
    'acknowledge',
    'version',
    'build',
  ],
  properties: {
    branch: {
      type: 'array',
    },
    region: {
      type: 'array',
    },
    jobService: {
      type: 'array',
    },
    jobServiceCompany: {
      type: 'array',
      items: {
        type: 'object',
        required: [
          'id',
          'jobCode',
          'jobDescription',
          'actual',
          'rate',
          'price',
          'vehicleType',
          'serviceCategory',
          'isActive',
        ],
        properties: {
          id: {
            type: 'integer',
          },
          jobCode: {
            type: 'string',
          },
          jobDescription: {
            type: 'string',
          },
          actual: {
            type: 'number',
          },
          rate: {
            type: 'integer',
          },
          price: {
            type: 'integer',
          },
          vehicleType: {
            type: 'string',
          },
          serviceCategory: {
            type: 'string',
          },
          isActive: {
            type: 'boolean',
          },
        },
      },
    },
    vehicleBrand: {
      type: 'array',
    },
    holidayCalendar: {
      type: 'array',
    },
    parameterConfigApplication: {
      type: 'array',
    },
    hobbies: {
      type: 'array',
    },
    province: {
      type: 'array',
    },
    district: {
      type: 'array',
    },
    vehicleTestDrive: {
      type: 'array',
    },
    location: {
      type: 'array',
    },
    acknowledge: {
      type: 'integer',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};

export const vehicleBrandSchema = {
  type: 'object',
  required: [
    'branch',
    'region',
    'jobService',
    'jobServiceCompany',
    'vehicleBrand',
    'holidayCalendar',
    'parameterConfigApplication',
    'hobbies',
    'province',
    'district',
    'vehicleTestDrive',
    'location',
    'acknowledge',
    'version',
    'build',
  ],
  properties: {
    branch: {
      type: 'array',
    },
    region: {
      type: 'array',
    },
    jobService: {
      type: 'array',
    },
    jobServiceCompany: {
      type: 'array',
    },
    vehicleBrand: {
      type: 'array',
      items: {
        type: 'object',
        required: [
          'id',
          'vehicleBrandCode',
          'vehicleType',
          'vehicleDescription',
          'isActive',
        ],
        properties: {
          id: {
            type: 'integer',
          },
          vehicleBrandCode: {
            type: 'string',
          },
          vehicleType: {
            type: 'string',
          },
          vehicleDescription: {
            type: 'string',
          },
          isActive: {
            type: 'boolean',
          },
        },
      },

    },
    holidayCalendar: {
      type: 'array',
    },
    parameterConfigApplication: {
      type: 'array',
    },
    hobbies: {
      type: 'array',
    },
    province: {
      type: 'array',
    },
    district: {
      type: 'array',
    },
    vehicleTestDrive: {
      type: 'array',
    },
    location: {
      type: 'array',
    },
    acknowledge: {
      type: 'integer',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};

export const masterDataProvinceSchema = {
  type: 'object',
  required: [
    'branch',
    'region',
    'jobService',
    'jobServiceCompany',
    'vehicleBrand',
    'holidayCalendar',
    'parameterConfigApplication',
    'hobbies',
    'province',
    'district',
    'vehicleTestDrive',
    'location',
    'acknowledge',
    'version',
    'build',
  ],
  properties: {
    branch: {
      type: 'array',
    },
    region: {
      type: 'array',
    },
    jobService: {
      type: 'array',
    },
    jobServiceCompany: {
      type: 'array',
    },
    vehicleBrand: {
      type: 'array',
    },

  },
  holidayCalendar: {
    type: 'array',
  },
  parameterConfigApplication: {
    type: 'array',
  },
  hobbies: {
    type: 'array',
  },
  province: {
    type: 'array',
    items: {
      type: 'object',
      required: [
        'id',
        'provinceCode',
        'provinceName',
        'isActive',
      ],
      properties: {
        id: {
          type: 'integer',
        },
        provinceCode: {
          type: 'string',
        },
        provinceName: {
          type: 'string',
        },
        isActive: {
          type: 'boolean',
        },
      },

    },
    district: {
      type: 'array',
    },
    vehicleTestDrive: {
      type: 'array',
    },
    location: {
      type: 'array',
    },
    acknowledge: {
      type: 'integer',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};

export const districtSchema = {
  type: 'object',
  required: [
    'branch',
    'region',
    'jobService',
    'jobServiceCompany',
    'vehicleBrand',
    'holidayCalendar',
    'parameterConfigApplication',
    'hobbies',
    'province',
    'district',
    'vehicleTestDrive',
    'location',
    'acknowledge',
    'version',
    'build',
  ],
  properties: {
    branch: {
      type: 'array',
    },
    region: {
      type: 'array',
    },
    jobService: {
      type: 'array',
    },
    jobServiceCompany: {
      type: 'array',
    },
    vehicleBrand: {
      type: 'array',

    },
    holidayCalendar: {
      type: 'array',
    },
    parameterConfigApplication: {
      type: 'array',
    },
    hobbies: {
      type: 'array',
    },
    province: {
      type: 'array',
    },
    district: {
      type: 'array',
      items: {
        type: 'object',
        required: [
          'id',
          'provinceId',
          'districtCode',
          'districtName',
          'isActive',
          'regions',
        ],
        properties: {
          id: {
            type: 'integer',
          },
          provinceId: {
            type: 'integer',
          },
          districtCode: {
            type: 'string',
          },
          districtName: {
            type: 'string',
          },
          isActive: {
            type: 'boolean',
          },
          regions: {
            type: 'array',
          },
        },
      },
    },
    vehicleTestDrive: {
      type: 'array',
    },
    location: {
      type: 'array',
    },
    acknowledge: {
      type: 'integer',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};

export const RegionsSchema = {
  type: 'array',
  items: {
    type: 'object',
    required: [
      'id',
      'salesOrganization',
      'regionCode',
      'regionName',
    ],
    properties: {
      id: {
        type: 'integer',
      },
      salesOrganization: {
        type: 'string',
      },
      regionCode: {
        type: 'string',
      },
      regionName: {
        type: 'string',
      },
    },
  },
};


export const masterDataRegionSchema = {
  type: 'object',
  required: [
    'branch',
    'region',
    'jobService',
    'jobServiceCompany',
    'vehicleBrand',
    'holidayCalendar',
    'parameterConfigApplication',
    'hobbies',
    'province',
    'district',
    'vehicleTestDrive',
    'location',
    'acknowledge',
    'version',
    'build',
  ],
  properties: {
    branch: {
      type: 'array',
    },
    region: {
      type: 'array',
      items: {
        type: 'object',
        required: [
          'id',
          'salesOrganization',
          'regionCode',
          'regionName',
        ],
        properties: {
          id: {
            type: 'integer',
          },
          salesOrganization: {
            type: 'string',
          },
          regionCode: {
            type: 'string',
          },
          regionName: {
            type: 'string',
          },
        },
      },
    },
    jobService: {
      type: 'array',
    },
    jobServiceCompany: {
      type: 'array',
    },
    vehicleBrand: {
      type: 'array',

    },
    holidayCalendar: {
      type: 'array',
    },
    parameterConfigApplication: {
      type: 'array',
    },
    hobbies: {
      type: 'array',
    },
    province: {
      type: 'array',
    },
    district: {
      type: 'array',
    },
    vehicleTestDrive: {
      type: 'array',
    },
    location: {
      type: 'array',
    },
    acknowledge: {
      type: 'integer',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};

export const holidayCalendarSchema = {
  type: 'object',
  required: [
    'branch',
    'region',
    'jobService',
    'jobServiceCompany',
    'vehicleBrand',
    'holidayCalendar',
    'parameterConfigApplication',
    'hobbies',
    'province',
    'district',
    'vehicleTestDrive',
    'location',
    'acknowledge',
    'version',
    'build',
  ],
  properties: {
    branch: {
      type: 'array',
    },
    region: {
      type: 'array',
    },
    jobService: {
      type: 'array',
    },
    jobServiceCompany: {
      type: 'array',
    },
    vehicleBrand: {
      type: 'array',

    },
    holidayCalendar: {
      type: 'array',
      items: {
        type: 'object',
        required: [
          'id',
          'holidayDate',
          'holidayDescription',
        ],
        properties: {
          id: {
            type: 'integer',
          },
          holidayDate: {
            type: 'string',
          },
          holidayDescription: {
            type: 'string',
          },
        },
      },
    },
    parameterConfigApplication: {
      type: 'array',
    },
    hobbies: {
      type: 'array',
    },
    province: {
      type: 'array',
    },
    district: {
      type: 'array',
    },
    vehicleTestDrive: {
      type: 'array',
    },
    location: {
      type: 'array',
    },
    acknowledge: {
      type: 'integer',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};

export const vehicleTestDriveSchema = {
  type: 'object',
  required: [
    'branch',
    'region',
    'jobService',
    'jobServiceCompany',
    'vehicleBrand',
    'holidayCalendar',
    'parameterConfigApplication',
    'hobbies',
    'province',
    'district',
    'vehicleTestDrive',
    'location',
    'acknowledge',
    'version',
    'build',
  ],
  properties: {
    branch: {
      type: 'array',
    },
    region: {
      type: 'array',
    },
    jobService: {
      type: 'array',
    },
    jobServiceCompany: {
      type: 'array',
    },
    vehicleBrand: {
      type: 'array',
    },
    holidayCalendar: {
      type: 'array',
    },
    parameterConfigApplication: {
      type: 'array',
    },
    hobbies: {
      type: 'array',
    },
    province: {
      type: 'array',
    },
    district: {
      type: 'array',
    },
    vehicleTestDrive: {
      type: 'array',
      items: {
        type: 'object',
        required: [
          'id',
          'vehicleBrandId',
          'vehicleType',
          'branchId',
          'branchCode',
          'isTestDrive',
          'isActive',
          'modelMaterialId',
        ],
        properties: {
          id: {
            type: 'integer',
          },
          vehicleBrandId: {
            type: 'integer',
          },
          vehicleType: {
            type: 'string',
          },
          branchId: {
            type: 'integer',
          },
          branchCode: {
            type: 'string',
          },
          isTestDrive: {
            type: 'boolean',
          },
          isActive: {
            type: 'boolean',
          },
          modelMaterialId: {
            type: 'integer',
          },
        },
      },
    },
    location: {
      type: 'array',
    },
    acknowledge: {
      type: 'integer',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};

export const parameterConfigApplicationSchema = {
  type: 'object',
  required: [
    'branch',
    'region',
    'jobService',
    'jobServiceCompany',
    'vehicleBrand',
    'holidayCalendar',
    'parameterConfigApplication',
    'hobbies',
    'province',
    'district',
    'vehicleTestDrive',
    'location',
    'acknowledge',
    'version',
    'build',
  ],
  properties: {
    branch: {
      type: 'array',
    },
    region: {
      type: 'array',
    },
    jobService: {
      type: 'array',
    },
    jobServiceCompany: {
      type: 'array',
    },
    vehicleBrand: {
      type: 'array',
    },
    holidayCalendar: {
      type: 'array',
    },
    parameterConfigApplication: {
      type: 'array',
      items: {
        type: 'object',
        required: [
          'id',
          'companyCode',
          'key',
          'value',
          'isActive',
        ],
        properties: {
          id: {
            type: 'integer',
          },
          companyCode: {
            type: 'string',
          },
          key: {
            type: 'string',
          },
          value: {
            type: 'string',
          },
          isActive: {
            type: 'boolean',
          },
        },
      },
    },
    hobbies: {
      type: 'array',
    },
    province: {
      type: 'array',
    },
    district: {
      type: 'array',
    },
    vehicleTestDrive: {
      type: 'array',
    },
    location: {
      type: 'array',
    },
    acknowledge: {
      type: 'integer',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};
