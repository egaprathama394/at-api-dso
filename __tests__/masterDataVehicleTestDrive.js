/* eslint-disable max-len */
/* eslint-disable eqeqeq */
/* eslint-disable no-shadow */
import getMasterData from '../endpoints/masterData';
import * as masterDataSchema from '../schemaValidator/masterData';
import * as body from '../data/masterData';

describe('Master Data Vehicle Test Drive', () => {
  test('Can get master data Vehicle test drive successfully - positive test cases', async () => {
    const response = await getMasterData(body.VehicleTestDrive);
    expect(response.status).toEqual(200);
    expect(response.body).toMatchSchema(masterDataSchema.vehicleTestDriveSchema);
    const vehicleTestDrive = await response.body.vehicleTestDrive;
    const result = vehicleTestDrive.filter(vehicleTestDrive => vehicleTestDrive.id != 0);
    expect(result.length).toEqual(15);
    expect(response.body).toMatchSnapshot();
  });
  test('Can get vehicle that eligible for test drive successfully - positive test cases', async () => {
    const response = await getMasterData(body.VehicleTestDrive);
    expect(response.status).toEqual(200);
    expect(response.body).toMatchSchema(masterDataSchema.vehicleTestDriveSchema);
    const vehicleTestDrive = await response.body.vehicleTestDrive;
    const result = vehicleTestDrive.filter(vehicleTestDrive => vehicleTestDrive.isTestDrive == true && vehicleTestDrive.isActive == true && vehicleTestDrive.modelMaterialId != 0);
    expect(result.length).toEqual(6);
    expect(response.body).toMatchSnapshot();
  });
});
