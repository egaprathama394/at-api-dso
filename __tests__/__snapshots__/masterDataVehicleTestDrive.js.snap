// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Master Data Vehicle Test Drive Can get master data Vehicle test drive successfully - positive test cases 1`] = `
Object {
  "acknowledge": 1,
  "branch": Array [],
  "build": "0",
  "district": Array [],
  "hobbies": Array [],
  "holidayCalendar": Array [],
  "jobService": Array [],
  "jobServiceCompany": Array [],
  "location": Array [],
  "parameterConfigApplication": Array [],
  "province": Array [],
  "region": Array [],
  "vehicleBrand": Array [],
  "vehicleTestDrive": Array [
    Object {
      "branchCode": "D004",
      "branchId": 262,
      "id": 1,
      "isActive": false,
      "isTestDrive": true,
      "modelMaterialId": 0,
      "vehicleBrandId": 77,
      "vehicleType": "TERIOS",
    },
    Object {
      "branchCode": "D004",
      "branchId": 262,
      "id": 2,
      "isActive": true,
      "isTestDrive": true,
      "modelMaterialId": 69,
      "vehicleBrandId": 224,
      "vehicleType": "AYLA A.T",
    },
    Object {
      "branchCode": "D004",
      "branchId": 262,
      "id": 3,
      "isActive": false,
      "isTestDrive": true,
      "modelMaterialId": 0,
      "vehicleBrandId": 87,
      "vehicleType": "GRANMAX1.3",
    },
    Object {
      "branchCode": "D004",
      "branchId": 262,
      "id": 4,
      "isActive": true,
      "isTestDrive": false,
      "modelMaterialId": 64,
      "vehicleBrandId": 88,
      "vehicleType": "GRANMAX1.5",
    },
    Object {
      "branchCode": "D004",
      "branchId": 262,
      "id": 5,
      "isActive": false,
      "isTestDrive": false,
      "modelMaterialId": 0,
      "vehicleBrandId": 93,
      "vehicleType": "LUXIO",
    },
    Object {
      "branchCode": "D004",
      "branchId": 262,
      "id": 6,
      "isActive": true,
      "isTestDrive": true,
      "modelMaterialId": 70,
      "vehicleBrandId": 225,
      "vehicleType": "SIGRA M.T",
    },
    Object {
      "branchCode": "D004",
      "branchId": 262,
      "id": 7,
      "isActive": true,
      "isTestDrive": true,
      "modelMaterialId": 70,
      "vehicleBrandId": 226,
      "vehicleType": "SIGRA A.T",
    },
    Object {
      "branchCode": "D004",
      "branchId": 262,
      "id": 8,
      "isActive": false,
      "isTestDrive": true,
      "modelMaterialId": 0,
      "vehicleBrandId": 90,
      "vehicleType": "SIRION",
    },
    Object {
      "branchCode": "D004",
      "branchId": 262,
      "id": 9,
      "isActive": true,
      "isTestDrive": true,
      "modelMaterialId": 72,
      "vehicleBrandId": 221,
      "vehicleType": "TERIOS A.T",
    },
    Object {
      "branchCode": "D004",
      "branchId": 262,
      "id": 10,
      "isActive": false,
      "isTestDrive": false,
      "modelMaterialId": 0,
      "vehicleBrandId": 91,
      "vehicleType": "XENIA1.0",
    },
    Object {
      "branchCode": "D004",
      "branchId": 262,
      "id": 11,
      "isActive": false,
      "isTestDrive": false,
      "modelMaterialId": 0,
      "vehicleBrandId": 92,
      "vehicleType": "XENIA1.3",
    },
    Object {
      "branchCode": "D005",
      "branchId": 1093,
      "id": 17,
      "isActive": true,
      "isTestDrive": false,
      "modelMaterialId": 0,
      "vehicleBrandId": 77,
      "vehicleType": "TERIOS",
    },
    Object {
      "branchCode": "D005",
      "branchId": 1093,
      "id": 20,
      "isActive": true,
      "isTestDrive": false,
      "modelMaterialId": 0,
      "vehicleBrandId": 123,
      "vehicleType": "Coba3",
    },
    Object {
      "branchCode": "D004",
      "branchId": 262,
      "id": 21,
      "isActive": true,
      "isTestDrive": true,
      "modelMaterialId": 72,
      "vehicleBrandId": 222,
      "vehicleType": "TERIOS M.T",
    },
    Object {
      "branchCode": "D004",
      "branchId": 262,
      "id": 22,
      "isActive": true,
      "isTestDrive": true,
      "modelMaterialId": 69,
      "vehicleBrandId": 223,
      "vehicleType": "AYLA M.T",
    },
  ],
  "version": "1.0",
}
`;

exports[`Master Data Vehicle Test Drive Can get vehicle that eligible for test drive successfully - positive test cases 1`] = `
Object {
  "acknowledge": 1,
  "branch": Array [],
  "build": "0",
  "district": Array [],
  "hobbies": Array [],
  "holidayCalendar": Array [],
  "jobService": Array [],
  "jobServiceCompany": Array [],
  "location": Array [],
  "parameterConfigApplication": Array [],
  "province": Array [],
  "region": Array [],
  "vehicleBrand": Array [],
  "vehicleTestDrive": Array [
    Object {
      "branchCode": "D004",
      "branchId": 262,
      "id": 1,
      "isActive": false,
      "isTestDrive": true,
      "modelMaterialId": 0,
      "vehicleBrandId": 77,
      "vehicleType": "TERIOS",
    },
    Object {
      "branchCode": "D004",
      "branchId": 262,
      "id": 2,
      "isActive": true,
      "isTestDrive": true,
      "modelMaterialId": 69,
      "vehicleBrandId": 224,
      "vehicleType": "AYLA A.T",
    },
    Object {
      "branchCode": "D004",
      "branchId": 262,
      "id": 3,
      "isActive": false,
      "isTestDrive": true,
      "modelMaterialId": 0,
      "vehicleBrandId": 87,
      "vehicleType": "GRANMAX1.3",
    },
    Object {
      "branchCode": "D004",
      "branchId": 262,
      "id": 4,
      "isActive": true,
      "isTestDrive": false,
      "modelMaterialId": 64,
      "vehicleBrandId": 88,
      "vehicleType": "GRANMAX1.5",
    },
    Object {
      "branchCode": "D004",
      "branchId": 262,
      "id": 5,
      "isActive": false,
      "isTestDrive": false,
      "modelMaterialId": 0,
      "vehicleBrandId": 93,
      "vehicleType": "LUXIO",
    },
    Object {
      "branchCode": "D004",
      "branchId": 262,
      "id": 6,
      "isActive": true,
      "isTestDrive": true,
      "modelMaterialId": 70,
      "vehicleBrandId": 225,
      "vehicleType": "SIGRA M.T",
    },
    Object {
      "branchCode": "D004",
      "branchId": 262,
      "id": 7,
      "isActive": true,
      "isTestDrive": true,
      "modelMaterialId": 70,
      "vehicleBrandId": 226,
      "vehicleType": "SIGRA A.T",
    },
    Object {
      "branchCode": "D004",
      "branchId": 262,
      "id": 8,
      "isActive": false,
      "isTestDrive": true,
      "modelMaterialId": 0,
      "vehicleBrandId": 90,
      "vehicleType": "SIRION",
    },
    Object {
      "branchCode": "D004",
      "branchId": 262,
      "id": 9,
      "isActive": true,
      "isTestDrive": true,
      "modelMaterialId": 72,
      "vehicleBrandId": 221,
      "vehicleType": "TERIOS A.T",
    },
    Object {
      "branchCode": "D004",
      "branchId": 262,
      "id": 10,
      "isActive": false,
      "isTestDrive": false,
      "modelMaterialId": 0,
      "vehicleBrandId": 91,
      "vehicleType": "XENIA1.0",
    },
    Object {
      "branchCode": "D004",
      "branchId": 262,
      "id": 11,
      "isActive": false,
      "isTestDrive": false,
      "modelMaterialId": 0,
      "vehicleBrandId": 92,
      "vehicleType": "XENIA1.3",
    },
    Object {
      "branchCode": "D005",
      "branchId": 1093,
      "id": 17,
      "isActive": true,
      "isTestDrive": false,
      "modelMaterialId": 0,
      "vehicleBrandId": 77,
      "vehicleType": "TERIOS",
    },
    Object {
      "branchCode": "D005",
      "branchId": 1093,
      "id": 20,
      "isActive": true,
      "isTestDrive": false,
      "modelMaterialId": 0,
      "vehicleBrandId": 123,
      "vehicleType": "Coba3",
    },
    Object {
      "branchCode": "D004",
      "branchId": 262,
      "id": 21,
      "isActive": true,
      "isTestDrive": true,
      "modelMaterialId": 72,
      "vehicleBrandId": 222,
      "vehicleType": "TERIOS M.T",
    },
    Object {
      "branchCode": "D004",
      "branchId": 262,
      "id": 22,
      "isActive": true,
      "isTestDrive": true,
      "modelMaterialId": 69,
      "vehicleBrandId": 223,
      "vehicleType": "AYLA M.T",
    },
  ],
  "version": "1.0",
}
`;
